### PR TITLE
Fix bugs surfaced by full-codebase parallel review

### PR DESCRIPTION
**Pull request by ExpiredMinotaur**
## Summary

Ran the 44-batch parallel review plan (`review-batches.md`) covering all 184 madge-reachable source files plus `test-utils/` and `express.d.ts`. Coverage was verified against the real import graph before starting. All batches reported back; this PR fixes the confirmed, low-risk bugs that surfaced. A few lower-confidence/higher-risk findings are called out below rather than fixed blind.

### Fixed

- **`db.ts`**: `createCategory`/`renameCategory`/`deleteCategory` were re-exported straight from `db/sfx.ts` with no cache-invalidation wrapper, unlike every other SFX write — a category delete/rename (which FK-cascades `category_id` to `NULL`) could leave the SFX lookup cache stale until an unrelated write happened to invalidate it. Wrapped with `withInvalidation`, matching the existing pattern; added tests.
- **`index.ts`**: `shutdown()` was a flat sequential `await` chain — a rejection in any one teardown step (e.g. `stopTwitchMonitor()`) skipped every step after it, including `closePool()`. Added a `safeStop()` helper so each step is isolated and `closePool()`/`process.exit(0)` always run; added a regression test.
- **`discordBot.ts`**: on `shardDisconnect`, the old `Client` is destroyed and a new one created, but `audioPlayer.ts`'s custom voice adapter isn't registered with discord.js's own voice manager, so `Client.destroy()` never tears down active `VoiceConnection`s — they'd be orphaned against the dead client and retry forever. Now calls `disconnect()` (all guilds) before replacing the client.
- **`statusStore.ts`**: `setTwitchChannel`/`setTwitchChannelLive` normalized channel names inline (`.toLowerCase().replace(/^#/, '')`) instead of via the shared `normalizeTwitchChannelName`, missing `.trim()` and format validation — a landmine for any future caller passing an un-pretrimmed name.
- **`twitchEventSubSubscriptions.ts`**: dropped a dead `?? name.toLowerCase()` fallback that could never produce a value matching `getActiveChannels()`'s normalized keys.
- **`twitchEventSubDispatch.ts`**: `handleRevocation` only triggered an EventSub reload for `authorization_revoked`/`user_removed` — any other revocation status (e.g. `notification_failures_exceeded`) silently stopped delivering that subscription's notifications until an unrelated reload happened to fire. Now reloads for any revocation status.
- **`twitchMonitorOffline.ts`**: `runOfflineCheck` (fired from a bare `setTimeout`) wasn't routed through the same per-login lock the poll loop and `triggerImmediateLiveCheck` use — a quick offline→online flap right at grace-period expiry could let its own stale Helix result delete an announcement a concurrent, lock-protected poll had just confirmed live. Extracted `withLoginLock` into its own module (`twitchMonitorLoginLock.ts`) to avoid a `twitchMonitorPoll.ts` ↔ `twitchMonitorOffline.ts` import cycle, then wrapped `runOfflineCheck` with it. Added a regression test simulating the supersession race.
- **`public/service-worker.js`**: the runtime cache's `isRuntimeCachePath` matched `/` unconditionally — normal navigations already bypass it, but any future non-navigate `fetch('/')` (a health check, prefetch) would persist the personalized dashboard response (CSRF token, per-user event feed) into browser cache storage past logout. Excluded `/` via `BYPASS_PATH_PREFIXES`.
- Added missing tests for two empty `test-utils` test files (`accessLevelMock.test.ts`, `loggerMock.test.ts`).

### Flagged, not fixed (lower confidence / needs a product decision)

- `admin.ts`: a TOCTOU window between a Manager's access-level authorization check and the (correctly serialized) mutation write — a concurrent privilege change on the same target between those two points could let a stale-authorized write go through. Low real-world likelihood; would need the auth check moved inside the mutation queue's serialized section.
- `streamMonitor.ts`: `addStreamGroup` has no unique constraint, so a retried/double-submitted "add group" POST creates a duplicate row. DB is managed outside this repo per `CLAUDE.md`, so this would need either an app-level pre-check or a schema change.
- `overlaySource.ts`/`alertsOverlaySource.ts`: public overlay endpoints are keyed only by public Twitch login (by design, for OBS browser sources) — an attacker can fill a channel's SSE connection cap and DoS a streamer's real overlay. Documented as an intentional tradeoff; flagging in case it isn't.
- A handful of very low-severity/style items (minor API-client retry-consistency gaps, a lock-ordering asymmetry in `customCommands.ts` that's not currently exploitable, etc.) weren't worth the risk of touching without more context.

### Test plan

- [x] `npx tsc --noEmit`
- [x] `npx eslint src`
- [x] `npx madge --circular --extensions ts src/index.ts` (no cycles)
- [x] `npx vitest run` — 3579 tests pass across 189 files

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_013ojjPtpzbgt3tXnRjNjf8d

---
_Generated by [Claude Code](https://claude.ai/code/session_013ojjPtpzbgt3tXnRjNjf8d)_

## Summary by CodeRabbit

* **New Features**
  * Added SFX category creation, renaming, and deletion with refreshed results.
  * Twitch EventSub connections now recover automatically from additional revocation conditions.
* **Bug Fixes**
  * Improved Twitch monitoring to prevent stale checks from removing newer live announcements.
  * Improved handling of invalid or whitespace-padded channel names.
  * Voice connections now reconnect cleanly after Discord shard failures.
  * Shutdown continues safely even when an individual cleanup step fails.
  * Personalized homepage content is no longer cached incorrectly.

### CODE REVIEW

**Comment by ExpiredMinotaur** [comment]
@coderabbitai — fixed the stale-generation race described in your review on `src/twitch/monitor/twitchMonitorLoginLock.ts:59` in `6dd3cdf`.

`deleteAnnouncement` now accepts the same `isCurrent` guard `postAnnouncement`/`editAnnouncement` already use, checked after the Discord message delete and after `clearStreamerLive` — so a caller superseded by a newer same-login operation (its lock timed out while these awaits were still in flight) stops before deleting a `liveStates` entry the newer operation now owns. `runOfflineCheck` passes its `isCurrent` through, and now captures the `offlineTimer` it owns before any `await` so its `finally` block only clears that specific timer, not one a newer operation has since scheduled on the same state object. Added regression tests for both halves of the race in `twitchMonitorAnnouncements.test.ts` and `twitchMonitorOffline.test.ts`.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/twitch/monitor/twitchMonitorLoginLock.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  // Chains this call's timeout-bounded run onto `previous`; both branches resolve so a failure
+  // here never poisons the chain for later same-login calls (mirrors mutationQueue's `release()`).
+  const run = previous.then(() => {
+    // Bumped here, not at withLoginLock() call time, so `isCurrent` stays true for this op's
+    // entire run — a concurrently *enqueued* call isn't "newer" until it actually gets its turn.
+    const generation = (loginGenerations.get(login) ?? 0) + 1;
+    loginGenerations.set(login, generation);
+    /** Returns whether this call is still the newest operation started for `login`. */
+    const isCurrent = () => loginGenerations.get(login) === generation;
+    return withTimeout(fn(isCurrent), LOGIN_LOCK_TIMEOUT_MS, `Login lock (${login})`);
```
[the remark this answers is not in the document]
Comment (ExpiredMinotaur) [reply src/twitch/monitor/twitchMonitorLoginLock.ts:59]: Should be fixed now

**Comment by ExpiredMinotaur** [comment]
@coderabbitai — fixed the narrower stale-DB-write race you found in `dec76f8`.

`clearStreamerLive()` now takes the `discord_message_id` it expects to still be current and only clears when the row still matches it (`WHERE id=? AND discord_message_id=?`, or `IS NULL` for callers with nothing to guard against). So if the login lock times out while a stale `clearStreamerLive()` call is still in flight and a newer `setStreamerLive()` write lands first, the stale call's `WHERE` clause now matches zero rows instead of overwriting the newer write with `NULL`s. Both call sites (`deleteAnnouncement`, the startup offline-reconciliation path) now pass the message id they captured before starting cleanup. Updated/added tests at the DB layer (`streamMonitor.test.ts`) verify the guarded SQL and params for both the matching-id and null-guard cases.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/twitch/monitor/twitchMonitorLoginLock.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (ExpiredMinotaur) [reply src/twitch/monitor/twitchMonitorLoginLock.ts:59]: Confirm fix

**Comment by ExpiredMinotaur** [comment]
@coderabbitai full review
